src/components/DonationCards/DonationCards.tsx

```diff
@@ -1,29 +1,75 @@
 import { useState, useEffect } from 'react';
 import styles from './DonationCards.module.css';
 import DonationCard from '../DonationCard';
+import DonationCardModal from '../DonationCardModal';
 import cards, { DonationCardData } from '../../data/cards';
 
 function DonationCards() {
    const [openCardId, setOpenCardId] = useState(-1);
+   const [isModalOpen, setIsModalOpen] = useState(false);
    const [donationCards, setDonationCards] = useState<DonationCardData[]>([]);
+   const [selectedCard, setSelectedCard] = useState<DonationCardData | null>(
+      null
+   );
 
    useEffect(() => {
       setDonationCards(cards);
    }, []);
 
+   const handleShowModal = (card: DonationCardData) => {
+      setSelectedCard(card);
+      setIsModalOpen(true);
+   };
+
+   const handleCloseModal = () => {
+      setIsModalOpen(false);
+      setSelectedCard(null);
+   };
+
+   const handleUpdateDonationAmount = (amount: number) => {
+      if (selectedCard) {
+         const cardIndex = donationCards.findIndex(
+            (card) => card.id === selectedCard.id
+         );
+         if (cardIndex !== -1) {
+            donationCards[cardIndex].donationAmount += amount;
+         }
+      }
+   };
+
    return (
       <main className={styles.donationCards}>
-         {donationCards.map(({ id, title, description, donationTarget }) => (
-            <DonationCard
-               key={id}
-               title={title}
-               description={description}
-               donationTarget={donationTarget}
-               amount={Math.random() * donationTarget}
-               isOpen={openCardId === id}
-               onShowMore={() => setOpenCardId(openCardId !== id ? id : -1)}
+         {donationCards.map(
+            ({ id, title, description, donationTarget, donationAmount }) => (
+               <DonationCard
+                  key={id}
+                  title={title}
+                  description={description}
+                  donationTarget={donationTarget}
+                  amount={donationAmount}
+                  isOpen={openCardId === id}
+                  onShowMore={() => setOpenCardId(openCardId !== id ? id : -1)}
+                  onShowModal={() =>
+                     handleShowModal({
+                        id,
+                        title,
+                        description,
+                        donationTarget,
+                        donationAmount,
+                     })
+                  }
+               />
+            )
+         )}
```

Comment:
Lets export it to a function that will return what is needed, so in the rendering we will see just stuff about rendering, so we will only see something like ```{renderDonationCard}```